delegation/src/main/java/com/iluwatar/delegation/simple/AbstractPrinterController.java

```diff
@@ -0,0 +1,34 @@
+package com.iluwatar.delegation.simple;
+
+/**
+ * Extra layer of abstraction for the controller to allow the controller in this example {@link PrinterController} to
+ * be as clean as possible. This just provides the default constructor and a simple getter method. The generic of
+ * T allows any implementation of {@link Printer}
+ *
+ * @param <T> Printer
+ * @see Printer
+ * @see PrinterController
+ */
+public abstract class AbstractPrinterController<T extends Printer> implements Printer {
```

Comment:
Type parameters don't bring any extra value here - you don't even use them. Remove generics and replace `T` with `Printer`.